src-ui/lib/components/BooksTableRow.svelte

```diff
@@ -1,15 +1,37 @@
 <script lang="ts">
+  import { createBooksStore } from '$lib/state/Books.svelte'
   import type { Book } from '$lib/types/book.js'
 
   export let book: Book
-  export let handleEdit: (book: Book, field: keyof Book, event: Event) => void
+
+  let booksStorePromise = createBooksStore()
+  const handleEdit = async (book: Book, field: keyof Book, e: Event) => {
+    const target = e.target as HTMLElement
+    const value =
+      field === 'authors'
+        ? target.innerText.split(',').map((author) => author.trim())
+        : target.innerText.trim()
+
+    return booksStorePromise.then(async (booksStore) => {
+      await booksStore.edit({ ...book, [field]: value })
+    })
+  }
+  const removeBook = async (id: string): Promise<void> => {
+    return booksStorePromise.then(async (booksStore) => {
+      await booksStore.remove(id)
+    })
+  }
 </script>
 
 <tr>
+  <td
+    ><button aria-label="delete book" class="delete" on:click={() => removeBook(book.id)}
+    ></button></td
+  >
   <td contenteditable="true" on:blur={(e) => handleEdit(book, 'isbn10', e)}
     >{book.isbn10 ?? book.isbn13}</td
   >
-  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'title', e)}>{book.title || ''} </td>
+  <td contenteditable="true" on:blur={(e) => handleEdit(book, 'title', e)}>{book?.title || ''} </td>
```

Comment:
why would book be `null`?

Reply:
Left over from trying to do a bug fix, will remove!